core/runtime/stats.rs

```diff
@@ -8,9 +8,62 @@ use crate::PromiseId;
 use crate::ResourceId;
 use bit_set::BitSet;
 use serde::Serialize;
+use serde::Serializer;
+use std::cell::Cell;
 use std::cell::RefCell;
+use std::collections::HashMap;
+use std::ops::Deref;
 use std::rc::Rc;
 
+#[derive(Default)]
+pub struct OpCallTraces {
+  enabled: Cell<bool>,
+  traces: RefCell<HashMap<PromiseId, Rc<str>>>,
```

Comment:
We need to keep this RefCell because the more efficient get/get_all might allow the user to call back into this struct. It's unlikely but possible.